pydantic/main.py

```diff
@@ -574,9 +577,25 @@ def create_model(  # noqa: C901 (ignore complexity)
     return type(model_name, (__base__,), namespace)  # type: ignore
 
 
+def _get_input_value(input_data: Any, key: str) -> Any:
+    try:
+        return input_data.get(key, _missing)
+    except AttributeError:
+        return getattr(input_data, key, _missing)
+
+
 def validate_model(  # noqa: C901 (ignore complexity)
-    model: Union[BaseModel, Type[BaseModel]], input_data: 'DictStrAny', raise_exc: bool = True, cls: 'ModelOrDc' = None
-) -> Union['DictStrAny', Tuple['DictStrAny', Optional[ValidationError]]]:
+    model: Union[BaseModel, Type[BaseModel]],
+    input_data: 'DictStrAny',
+    raise_exc: bool = True,
+    return_fields_set: bool = False,
```

Comment:
surely we can make `validate_model` always return `fields_set`? see https://github.com/samuelcolvin/pydantic/pull/518#discussion_r283133298

Reply:
Cool! That's easier/simpler.

Reply:
I implemented it, and I just included a fix for #517 (removing 2 lines). I tested it locally with @sommd's tests implemented in PR #518 and they passed.

I didn't include his tests in this PR because I think he deserves the credit for those tests, for finding the bug and writing the tests for it.

Reply:
agreed, I'll try to merge #518 before this PR so you might get some conflicts.

Reply:
No prob, I'll fix them.

Still, if this PR is accepted, the implementation of #518 would change (except the tests).

If you merge this first, and then #518 only adds the tests, there wouldn't be any conflicts (as I'm intentionally not adding those tests).

Or you can copy the code in here to #518 (or some part of it).